### PR TITLE
Added CBMC proof of CleanupData_Http function

### PR DESCRIPTION
**Pull request by Bujain**
*Description of changes:*
Added the files required for CBMC Proof of CleanupData_Http function.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/cbmc/proofs/Makefile-template-defines [hunk 1]
```diff
@@ -1,6 +1,7 @@
 
 # Absolute path to the root of the source tree.
 #
+
```
Comment (yanjos-dev) [inline test/cbmc/proofs/Makefile-template-defines:4]: If this newline is added by accident, please remove it.

File: test/cbmc/proofs/cleanupData_Http/cleanupData_Http_harness.c [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    OtaHttpStatus_t status;
+
+    OtaAgentContext_t agent;
+    OtaInterfaces_t interface;
+
+    pAgentCtx = &agent;
+    pInterfaces = &interface;
+
+    /* This assumption is made because the deinit function can never be NULL
+     *  It is always pointing to a function. */
```
Comment (yanjos-dev) [inline test/cbmc/proofs/cleanupData_Http/cleanupData_Http_harness.c:53]: Please update this comment to be a complete sentence.

File: test/cbmc/proofs/cleanupData_Http/cleanupData_Http_harness.c [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    pHttp.deinit = deinit;
+
+    /* Update the interface and the Agent. */
```
Comment (yanjos-dev) [inline test/cbmc/proofs/cleanupData_Http/cleanupData_Http_harness.c:56]: This comment is unnecessary, you can remove it.

File: test/cbmc/proofs/cleanupData_Http/cleanupData_Http_harness.c [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+}
+/*-----------------------------------------------------------*/
+
+void cleanupData_Http_harness()
+{
+    OtaAgentContext_t * pAgentCtx;
+    OtaHttpInterface_t pHttp;
+    OtaInterfaces_t * pInterfaces;
+    OtaHttpStatus_t status;
+
```
Comment (yanjos-dev) [inline test/cbmc/proofs/cleanupData_Http/cleanupData_Http_harness.c:45]: Can you remove this extra whitespace?

**Review by yanjos-dev — CHANGES_REQUESTED** [review]

File: test/cbmc/proofs/Makefile-template-defines (hunk 1 above)
Comment (Bujain) [reply test/cbmc/proofs/Makefile-template-defines]: Updated it in 8f0b7b2

File: test/cbmc/proofs/cleanupData_Http/cleanupData_Http_harness.c (hunk 2 above)
Comment (Bujain) [reply test/cbmc/proofs/cleanupData_Http/cleanupData_Http_harness.c]: Updated it in 8f0b7b2

File: test/cbmc/proofs/cleanupData_Http/cleanupData_Http_harness.c (hunk 3 above)
Comment (Bujain) [reply test/cbmc/proofs/cleanupData_Http/cleanupData_Http_harness.c]: Updated it in 8f0b7b2

File: test/cbmc/proofs/cleanupData_Http/cleanupData_Http_harness.c (hunk 4 above)
Comment (Bujain) [reply test/cbmc/proofs/cleanupData_Http/cleanupData_Http_harness.c]: Updated it in 8f0b7b2

File: test/cbmc/proofs/cleanupData_Http/cleanupData_Http_harness.c [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    pInterfaces->http = pHttp;
+    pAgentCtx->pOtaInterface = pInterfaces;
+
+    status = cleanupData_Http( pAgentCtx );
+
+    /* The function should either return OtaErrNone or OtaErrCleanupDataFailed. */
+    __CPROVER_assert( ( status == OtaErrNone ) ||
+                      ( status == OtaErrCleanupDataFailed ),
+                      "The function return should be either OtaErrNone or OtaErrCleanupDataFailed" );
```
Comment (yanjos-dev) [inline test/cbmc/proofs/cleanupData_Http/cleanupData_Http_harness.c:63]: Please update this error message based on our previous offline discussion.

**Review by yanjos-dev — CHANGES_REQUESTED** [review]

File: test/cbmc/proofs/cleanupData_Http/cleanupData_Http_harness.c (hunk 5 above)
Comment (Bujain) [reply test/cbmc/proofs/cleanupData_Http/cleanupData_Http_harness.c]: Updated it in  1c126b2

**Review by yanjos-dev — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**